### PR TITLE
Rationalize test environments to improve build speed.

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary

Currently we run Python build tests, license checks, and Python linting in every test environment.

This means we are testing the same thing repeatedly for no good reason, and also if there is a failure, then it will happen repeatedly across many test environments, which reduces the diagnosticity of the tests.

This PR separates out each of these things into their own tox testenv, to give better encapsulation of these tests and to improve build times.

## TODO

- [X] Have tests been written for the new code?

### CODE REVIEW

**Comment by rtibbles** [comment]
This is now ready to go.

Build time reported by Travis for successful build: 30 min 28 sec.

Significant improvement!

**Comment by rtibbles** [comment]
Rebased to master again to resolve merge conflict.

File: .travis.yml [hunk 1]
```diff
@@ -16,6 +16,26 @@ matrix:
...
+        - TRAVIS_NODE_VERSION="0.12"
+    - python: "3.4"
+      env:
+        - TOX_ENV=pythonbuild3.4
+        - TRAVIS_NODE_VERSION="0.12"
+    - python: "3.5"
+      env:
+        - TOX_ENV=pythonbuild3.5
+        - TRAVIS_NODE_VERSION="0.12"
+    - python: "2.7"
```
Comment (ralphiee22) [inline .travis.yml:29]: Maybe we should also be linting for 3.5 atleast. Not too familiar with discrepancies in the versions but `print 'hello'` and `print('hello')` comes to mind.

**Comment by rtibbles** [comment]
Opened an issue for this: #551

**Pull request merged**